modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -873,25 +873,25 @@ private void initDataNodesFromVaultManager() {
                         try {
                             if (vaultEntry != null && vaultEntry.value() != null) {
                                 logicalTopology = fromBytes(vaultEntry.value());
+                            }
 
-                                // init keys and data nodes for default zone
-                                saveDataNodesAndUpdateTriggerKeysInMetaStorage(
-                                        DEFAULT_ZONE_ID,
-                                        appliedRevision,
-                                        logicalTopology
-                                );
+                            // init keys and data nodes for default zone
+                            saveDataNodesAndUpdateTriggerKeysInMetaStorage(
+                                    DEFAULT_ZONE_ID,
+                                    appliedRevision,
+                                    logicalTopology
```

Comment:
Are we going to call saveDataNodesAndUpdateTriggerKeysInMetaStorage if vaultEntry == null or  vaultEntry.value() == null? If true
a. Why?
b. What we will use as logicalTopology?

Reply:
I don't agree. What's the point in writing empty dataNodes and fake appliedRevision? That should be handled within saveDataNodesToMetaStorageOn...

Reply:
It is easier to initialize it once than to do null checks in saveDataNodesToMetaStorageOnScaleUp and saveDataNodesToMetaStorageOnScaleDown and may be later in other places

Reply:
Well, the easiest way will be to catch NPE and ignore it. What I wan't to say here is that easier doesn't mean better.

Reply:
Ok. Fixed.